.gitignore

```diff
@@ -1,2 +1,17 @@
+# OS
+.DS_Store
+*.swp
+*.swo
+*.swl
+*.swm
+*.swn
+*.pyc
+
 target/
 Cargo.lock
+
+# IDE
+.idea
+*.iml
+.dir-locals.el
+.vscode
```

Comment:
I would suggest configuring your local .gitconfig with:

```
[core]
  excludesfile = /home/me/.gitignore
```

That way you can globally define what editor-specific artifacts you would like ignored, rather than trying to upstream this to every `.gitignore` file you interact with (and in turn, those `.gitignore` files having to accommodate things like swap files from every IDE under the sun)